src/main/scala/io/shiftleft/fuzzyc2cpg/FuzzyC2Cpg.scala

```diff
@@ -113,6 +127,52 @@ class FuzzyC2Cpg(outputModuleFactory: CpgOutputModuleFactory) {
 
 }
 
+object FuzzyC2CpgCache {
+  private val functionDeclarations = new mutable.HashMap[String, mutable.ListBuffer[(String, CpgStruct.Builder)]]()
+
+  /**
+    * Unless `remove` has been called for `signature`, add (outputIdentifier, cpg)
+    * pair to the list declarations stored for `signature`.
+    * */
+  def add(signature: String, outputIdentifier: String, cpg: CpgStruct.Builder): Unit = {
+    functionDeclarations.synchronized {
+      if (functionDeclarations.contains(signature)) {
+        val declList = functionDeclarations(signature)
+        if (declList.nonEmpty) {
+          functionDeclarations(signature).append((outputIdentifier, cpg))
```

Comment:
Can we reuse `declList` here?

```suggestion
          declList.append((outputIdentifier, cpg))
```

Reply:
good catch, done.